ingestion/src/metadata/profiler/orm/converter/base.py

```diff
@@ -198,3 +211,22 @@ def get_orm_database(table: Table, metadata: OpenMetadata) -> str:
     )
 
     return str(database.name.root)
+
+
+class ORMTableRegsitry:
+    # we store the ORM table in a registry to avoid rebuilding the same table. This is required becuase
+    # when using the same table in multiple places, we need to ensure that the ORM object is the same or
+    # the query builder will generate JOIN clauses that are not valid.
+    registry: Dict[str, DeclarativeMeta] = {}
+
+    def build_table_orm(
+        self, table: Table, service_conn_config: BaseModel, ometa_client: OpenMetadata
+    ) -> Optional[DeclarativeMeta]:
+        """Build the ORM table if needed for the sampler and profiler interfaces"""
+        if table.fullyQualifiedName.root in self.registry:
+            return self.registry[table.fullyQualifiedName.root]
+        if service_conn_config.type.value not in NON_SQA_DATABASE_CONNECTIONS:
+            orm_obj = ometa_to_sqa_orm(table, ometa_client)
+            self.registry[table.fullyQualifiedName.root] = orm_obj
+            return orm_obj
+        return None
```

Comment:
This used to be in the interface SQAMixIn. This is SQA specific, if we are going to move it again, let's move it back to the MixIn IMO where all SQA ops live.

Reply:
This requires a global state. do you want to add it to the SQA mixin? IMO the SQAMixin is becoming a [bloater](https://refactoring.guru/refactoring/smells/bloaters) by capturing all functionality related to to SQA and it should be broken down to more bit-sizes pieces.

If you have a concrete example of how to implement this in the `SQAMixin` along with the state please share so I can see what you mean.

Reply:
moved to `SQAMixin`